src/Compilers/CSharp/Test/Semantic/Semantics/RecordTests.cs

```diff
@@ -26967,5 +26967,109 @@ public class Outer
             CompileAndVerify(compRelease, expectedOutput: "C1 { I1 = 42 }", verify: Verification.Skipped /* init-only */);
             compRelease.VerifyEmitDiagnostics();
         }
+
+        [Fact]
+        [WorkItem(44571, "https://github.com/dotnet/roslyn/issues/44571")]
+        public void XmlDoc()
+        {
+            var src = @"
+/// <summary>Summary</summary>
+/// <param name=""I1"">Description for I1</param>
+public record C(int I1);
+
+namespace System.Runtime.CompilerServices
+{
+    /// <summary>Ignored</summary>
+    public static class IsExternalInit
+    {
+    }
+}
+";
+
+            var comp = CreateCompilation(src, parseOptions: TestOptions.RegularWithDocumentationComments);
+
+            var tree = comp.SyntaxTrees[0];
+            var model = comp.GetSemanticModel(tree, ignoreAccessibility: false);
+            var parameter = tree.GetRoot().DescendantNodes().OfType<ParameterSyntax>().First();
+            var i1 = model.GetDeclaredSymbol(parameter);
+
+            comp.VerifyDiagnostics(
+                // (4,21): warning CS1591: Missing XML comment for publicly visible type or member 'C.I1'
+                // public record C(int I1);
+                Diagnostic(ErrorCode.WRN_MissingXMLComment, "I1").WithArguments("C.I1").WithLocation(4, 21)
+                );
+
+            var cMember = comp.GetMember<NamedTypeSymbol>("C");
+            Assert.Equal(
+@"<member name=""T:C"">
+    <summary>Summary</summary>
+    <param name=""I1"">Description for I1</param>
+</member>
+", cMember.GetDocumentationCommentXml());
+
+            var constructor = cMember.GetMembers(".ctor").First();
+            Assert.Equal(
+@"<member name=""M:C.#ctor(System.Int32)"">
+    <summary>Summary</summary>
+    <param name=""I1"">Description for I1</param>
+</member>
+", constructor.GetDocumentationCommentXml());
```

Comment:
>GetDocumentationCommentXml [](start = 15, length = 26)

I didn't see any special code change to add this doc-comment to the constructor. How does this work?

Reply:
The declaring syntax for the constructor is the record declaration, so we find the proper trivia containing xmldocs.

Reply:
>The declaring syntax for the constructor is the record declaration, so we find the proper trivia containing xmldocs.

So, we attach types doc-comment to the constructor? I do not think we made an explicit decision to do that. It works this way by accident.  

---
In reply to: [518263269](https://github.com/dotnet/roslyn/pull/49134#discussion_r518263269) [](ancestors = 518263269)

Reply:
I think we want the xmldoc on the record to apply to the contructor too, that's consistent with wanting the `<param>` to apply to the primary constructor parameters.
Can you clarify what is your expectation or proposal?

Reply:
> I think we want the xmldoc on the record to apply to the contructor too, that's consistent with wanting the <param> to apply to the primary constructor parameters.
Can you clarify what is your expectation or proposal? 

I believe, we decided to transfer content of ```<param>``` to the constructor, not the entire doc comment. I am proposing to either explicitly make this decision (outside of this PR), or to follow previous decision by the letter. I.e. transfer the tags rather than the entire doc-comment.

---
In reply to: [518290263](https://github.com/dotnet/roslyn/pull/49134#discussion_r518290263) [](ancestors = 518290263)

Reply:
Confirmed expectations by email with LDM. Notes from 11/9/2020:
1. The entire xml doc from the record applies to the primary constructor (not just <param>)
2. Missing xml doc warning is suppressed on synthesized properties for now


---
In reply to: [518293736](https://github.com/dotnet/roslyn/pull/49134#discussion_r518293736) [](ancestors = 518293736,518290263)